python/ray/util/spark/cluster_init.py

```diff
@@ -380,83 +384,33 @@ def _init_ray_cluster(
         )
-
-    insufficient_resources = []
-
-    if num_spark_task_cpus < 4:
-        insufficient_resources.append(
-            "The provided CPU resources for each ray worker are inadequate to start "
-            "a ray cluster. Based on the total cpu resources available and the "
-            "configured task sizing, each ray worker would start with "
-            f"{num_spark_task_cpus} CPU cores. This is less than the recommended "
-            "value of `4` CPUs per worker. Increasing the spark configuration "
-            "'spark.task.cpus' to a minimum of `4` addresses it."
+        ray_dashboard_agent_port = get_random_unused_port(
+            ray_head_ip,
+            min_port=9000,
+            max_port=10000,
+            exclude_list=[ray_head_port, ray_dashboard_port],
         )
-
-    if ray_worker_node_heap_mem_bytes < 10 * 1024 * 1024 * 1024:
-        insufficient_resources.append(
-            "The provided memory resources for each ray worker are inadequate. Based "
-            "on the total memory available on the spark cluster and the configured "
-            "task sizing, each ray worker would start with "
-            f"{ray_worker_node_heap_mem_bytes} bytes heap memory. This is less than "
-            "the recommended value of 10GB. The ray worker node heap memory size is "
-            "calculated by "
-            "(SPARK_WORKER_NODE_PHYSICAL_MEMORY / num_local_spark_task_slots * 0.8) - "
-            "object_store_memory_per_node. To increase the heap space available, "
-            "increase the memory in the spark cluster by changing instance types or "
-            "worker count, reduce the target `num_worker_nodes`, or apply a lower "
-            "`object_store_memory_per_node`."
-        )
-    if insufficient_resources:
-        if safe_mode:
-            raise ValueError(
-                "You are creating ray cluster on spark with safe mode (it can be "
-                "disabled by setting argument 'safe_mode=False' when calling API "
-                "'init_ray_cluster'), safe mode requires the spark cluster config "
-                "satisfying following criterion: "
-                "\n".join(insufficient_resources)
-            )
-        else:
-            _logger.warning("\n".join(insufficient_resources))
-
-    ray_head_ip = socket.gethostbyname(get_spark_application_driver_host(spark))
-
-    ray_head_port = get_random_unused_port(ray_head_ip, min_port=9000, max_port=10000)
-    ray_dashboard_port = get_random_unused_port(
-        ray_head_ip, min_port=9000, max_port=10000, exclude_list=[ray_head_port]
-    )
-    ray_dashboard_agent_port = get_random_unused_port(
-        ray_head_ip,
-        min_port=9000,
-        max_port=10000,
-        exclude_list=[ray_head_port, ray_dashboard_port],
-    )
+        dashboard_options = {
+            "--include-dashboard=true",
+            "--dashboard-host=0.0.0.0",
+            f"--dashboard-port={ray_dashboard_port}",
```

Comment:
what if user wants to specify their dashboard port through `head_node_options`? IIRC, Ray will find a free port for you automatically.

Reply:
To avoid port conflicts with other ray services (e.g. conflicts with Ray worker port range if user run on locally deployed spark), I restrict the range of  ray_dashboard_port, 
also, Databricks runtime has restriction on port range that can be forwarded by driver proxy.

So I don't allow user to specify their own ray_dashboard_port but internally generate it.

Reply:
Allow user to set ray_dashboard_port sounds like a reasonable request. Some user might want to run dashboard on specific port. I can change it.